File: backend/app/schemas/task.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum

class TaskStatus(str, Enum):
    pending = "pending"
    in_progress = "in_progress"
    completed = "completed"

class TaskPriority(str, Enum):
    low = "low"
    medium = "medium"
    high = "high"

class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: TaskStatus = TaskStatus.pending
    priority: TaskPriority = TaskPriority.medium
    due_date: Optional[datetime] = None
    tags: List[str] = []

class TaskCreate(TaskBase):
    user_id: str

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[TaskStatus] = None
    priority: Optional[TaskPriority] = None
    due_date: Optional[datetime] = None
    tags: Optional[List[str]] = None
    column_order: Optional[int] = None

class TaskInDB(TaskBase):
    id: str
    user_id: str
    column_order: int = 0
    created_at: datetime
    updated_at: Optional[datetime] = None
    is_deleted: bool = False
    related_goal_id: Optional[str] = None
    category: Optional[str] = None

class Task(TaskInDB):
    pass 

File: frontend/src/types/task.ts
/**
 * Tipos relacionados con tareas
 */

export interface Task {
  id: string;
  user_id: string;
  title: string;
  description?: string;
  status: 'pending' | 'in_progress' | 'completed' | 'canceled';
  priority: 'low' | 'medium' | 'high';
  due_date?: string;
  column_order?: number;
  created_at: string;
  updated_at: string;
  is_deleted?: boolean;
  related_goal_id?: string;
  category?: string;
  tags?: string[];
  estimated_time_minutes?: number;
  completed_at?: string;
}

export interface TaskCreateDTO {
  title: string;
  description?: string;
  status?: 'pending' | 'in_progress' | 'completed';
  priority?: 'low' | 'medium' | 'high';
  due_date?: string;
  related_goal_id?: string;
  category?: string;
  tags?: string[];
  estimated_time_minutes?: number;
}

export interface TaskUpdateDTO {
  title?: string;
  description?: string;
  status?: 'pending' | 'in_progress' | 'completed' | 'canceled';
  priority?: 'low' | 'medium' | 'high';
  due_date?: string | null;
  column_order?: number;
  related_goal_id?: string | null;
  category?: string;
  tags?: string[];
  estimated_time_minutes?: number | null;
}

export interface TaskFilters {
  status?: ('pending' | 'in_progress' | 'completed')[];
  priority?: ('low' | 'medium' | 'high')[];
  dueDate?: 'today' | 'tomorrow' | 'this_week' | 'next_week' | 'this_month' | 'overdue' | 'no_date';
  category?: string[];
  tags?: string[];
  search?: string;
  relatedGoalId?: string;
}

export interface TaskSortOptions {
  field: 'due_date' | 'priority' | 'created_at' | 'title';
  direction: 'asc' | 'desc';
} 